### PR TITLE
adding fairlearn

### PR DESCRIPTION
**Pull request by adrinjalali**
Adding the recipe for [fairlearn](https://github.com/fairlearn/fairlearn)

ping  @romanlutz, @riedgar-ms

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by adrinjalali** [comment]
The CI now fails due to:

        error: can't copy 'fairlearn/widget/static/extension.js': doesn't exist or not a regular file

Not sure how to handle the extension. Ideally it shouldn't be in this package if you ask me. As a user, I should be able to use the package w/o a notebook and the widget.

WDYT @romanlutz

File: recipes/fairlearn/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{% set version = "0.4.0" %}
```
Comment (romanlutz) [inline recipes/fairlearn/meta.yaml]: Does that mean we have to keep updating this with every new version?

Comment (adrinjalali) [reply recipes/fairlearn/meta.yaml]: a bot does that automatically whenever you publish a new release on github, for example: https://github.com/conda-forge/scikit-learn-feedstock/pull/110

Or at least, I know there is a way for it to do that :D

**Comment by riedgar-ms** [comment]
> The CI now fails due to:
>     error: can't copy 'fairlearn/widget/static/extension.js': doesn't exist or not a regular file
> 
> Not sure how to handle the extension. Ideally it shouldn't be in this package if you ask me. As a user, I should be able to use the package w/o a notebook and the widget.
> WDYT @romanlutz

Is this a problem from the `pip install` or a `git clone`? If the latter, it's expected, since there's a `yarn` build step in the release to PyPI. See:
https://github.com/fairlearn/fairlearn#install-fairlearn

Having said that, we have started thinking about having a 'core' package which ditches all the UX bits.

**Comment by adrinjalali** [comment]
No it's a packaging issue: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=107632&view=logs&j=10bda42a-d420-5c28-a43f-bc4e66405e5c&t=bef775d0-a1a2-55a3-9721-17147029baf0&l=975

I like the core package idea :)

**Comment by riedgar-ms** [comment]
> No it's a packaging issue: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=107632&view=logs&j=10bda42a-d420-5c28-a43f-bc4e66405e5c&t=bef775d0-a1a2-55a3-9721-17147029baf0&l=975
> I like the core package idea :)

I wonder why none of our pipelines see this. I'd need to talk to Brandon to find out what that file does.

File: recipes/fairlearn/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+package:
+  name: fairlearn
+  version: {{ version }}
+
+source:
+  url: https://github.com/fairlearn/fairlearn/archive/v{{ version }}.tar.gz
+  sha256: 38ee80303ede1cddf119cbc0bf08f337978103f0e90badef97c64087fd4613f2
+  # sha256 is the preferred checksum -- you can get it for a file with:
```
Comment (hmaarrfk) [inline recipes/fairlearn/meta.yaml]: THese commens can go.

File: recipes/fairlearn/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  #  `openssl sha256 <file name>`.
+  # You may need the openssl package, available on conda-forge:
+  #  `conda install openssl -c conda-forge``
+
+build:
+  number: 0
+
+  # If the installation is complex, or different between Unix and Windows, use
```
Comment (hmaarrfk) [inline recipes/fairlearn/meta.yaml]: This comment can go too.

File: recipes/fairlearn/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  # separate bld.bat and build.sh files instead of this key. By default, the
+  # package will be built for the Python versions supported by conda-forge and
+  # for all major OSs. Add the line "skip: True # [py<35]" (for example) to
+  # limit to Python 3.5 and newer, or "skip: True # [not win]" to limit to
+  # Windows.
+
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: True  # [py<35]
```
Comment (hmaarrfk) [inline recipes/fairlearn/meta.yaml]: This seems like a noarch package to me. Is anything version specific?
https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=noarch#noarch-python

File: recipes/fairlearn/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    - fairlearn
+
+about:
+  home: https://github.com/fairlearn/fairlearn
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: 'Simple and easy fairness assessment and unfairness mitigation'
+
+  # The remaining entries in this section are optional, but recommended.
```
Comment (hmaarrfk) [inline recipes/fairlearn/meta.yaml]: You can remove this comment

File: recipes/fairlearn/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  description: |
+    The fairlearn project seeks to enable anyone involved in the development of
+    artificial intelligence (AI) systems to assess their system's fairness and
+    mitigate the observed unfairness.
+
```
Comment (hmaarrfk) [inline recipes/fairlearn/meta.yaml]: reduced spacing?

File: recipes/fairlearn/meta.yaml (hunk 4 above)
Comment (adrinjalali) [reply recipes/fairlearn/meta.yaml]: I'd agree, but linter disagreed: https://github.com/conda-forge/staged-recipes/pull/10518#issuecomment-570553673

**Comment by adrinjalali** [comment]
Thanks @hmaarrfk , for the failing CIs, I guess we'll wait for the team to do a release w/o the UI related components, and then ti should all go green.

**Comment by hmaarrfk** [comment]
Why not add the UI dependencies?

File: recipes/fairlearn/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+source:
+  url: https://github.com/fairlearn/fairlearn/archive/v{{ version }}.tar.gz
+  sha256: f602f572aa1aee36709d9edb38f5ff77c606a64caa92a8219328f6a127ec5411
+
+build:
+  number: 0
+
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: True  # [py<35]
```
Comment (hmaarrfk) [inline recipes/fairlearn/meta.yaml]:
```suggestion
  skip: True
```

File: recipes/fairlearn/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+requirements:
+  host:
+    - python
```
Comment (hmaarrfk) [inline recipes/fairlearn/meta.yaml]:
```suggestion
    - python >=3.6
```

File: recipes/fairlearn/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    - pip
+  run:
+    - python
```
Comment (hmaarrfk) [inline recipes/fairlearn/meta.yaml]:
```suggestion
    - python >=3.6
```

**Comment by adrinjalali** [comment]
@hmaarrfk thanks for the review, but your suggestions are triggering a liter error and I'm not sure how to fix it:

> Non noarch: python packages should have a python requirement without any version constraints.

**Comment by hmaarrfk** [comment]
Remove the skip statement all together

File: recipes/fairlearn/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+requirements:
+  host:
+    - python >=3.5
+    - pip
+  run:
+    - python >=3.5
+
+test:
+  # Some packages might need a `test/commands` key to check CLI.
```
Comment (hmaarrfk) [inline recipes/fairlearn/meta.yaml]: You should remove these comments

**Comment by hmaarrfk** [comment]
I don't get what the longer is complaining about. This looks good to me.

File: recipes/fairlearn/meta.yaml (hunk 10 above)
Comment (adrinjalali) [reply recipes/fairlearn/meta.yaml]: I had them there for us to remember what they are ;) removed now.

**Comment by hmaarrfk** [comment]
@conda-forge/staged-recipes this looks good for merge

**Comment by npavlovikj** [comment]
This recipe should be `noarch`. Non-noarch recipes shouldn't have version constraints, thus the linting message.

File: recipes/fairlearn/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+  name: fairlearn
+  version: {{ version }}
+
+source:
+  url: https://github.com/fairlearn/fairlearn/archive/v{{ version }}.tar.gz
+  sha256: f602f572aa1aee36709d9edb38f5ff77c606a64caa92a8219328f6a127ec5411
+
+build:
+  number: 0
+
```
Comment (chrisburr) [inline recipes/fairlearn/meta.yaml]:
```suggestion
  noarch: python
```

**Comment by adrinjalali** [comment]
All green, gentle ping, and thank you all for the help and the awesome work :)

**Pull request merged**